x/dsmr/node.go

```diff
@@ -46,6 +47,13 @@ var (
 	ErrFailedToReplicate                   = errors.New("failed to replicate to sufficient stake")
 )
 
+type Interface[T Tx] interface {
+	BuildChunk(ctx context.Context, txs []T, expiry int64, beneficiary codec.Address) error
+	BuildBlock(parent Block, timestamp int64) (Block, error)
+	Verify(ctx context.Context, parent Block, block Block) error
+	Accept(ctx context.Context, block Block) (ExecutedBlock[T], error)
+}
```

Comment:
What's the rationale for defining the interface this way? Asking because we've previously discussed removing defining interfaces over a single implementation

Reply:
I thought about this last night and originally felt this made sense so that we could keep wrapping `dsmr` with more features, but in hindsight maybe it makes more sense for this to live in the `fdsmr` package and say that `dsmr` implements it (i.e `dsmr` is fortifiable).

Another idea is to make `fdsmr` rely on the concrete type of `dsmr` since we only have one implementation but this makes testing look weirder.